### PR TITLE
add explicit pull for local serve

### PR DESCRIPTION
**Pull request by ChoiByungWook**
Explicit pull for ECR image, as serve can potentially timeout due to ping.

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have updated the [changelog](https://github.com/aws/sagemaker-python-sdk/blob/master/CHANGELOG.rst) with a description of my changes (if appropriate)
- [x] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/local/image.py [hunk 1]
```diff
@@ -668,3 +668,13 @@ def _ecr_login_if_needed(boto_session, image):
 
     cmd = "docker login -u AWS -p %s %s" % (token, ecr_url)
     subprocess.check_output(cmd, shell=True)
+
+    return True
+
+
+def _pull_image(image):
+    pull_image_command = ('docker pull %s' % image).strip()
+    print('docker command: {}'.format(pull_image_command))
```
Comment (iquintero) [inline src/sagemaker/local/image.py]: Shouldn't this be done through the logger?

File: src/sagemaker/local/image.py [hunk 2]
```diff
@@ -668,3 +671,13 @@ def _ecr_login_if_needed(boto_session, image):
 
     cmd = "docker login -u AWS -p %s %s" % (token, ecr_url)
     subprocess.check_output(cmd, shell=True)
+
+    return True
```
Comment (iquintero) [inline src/sagemaker/local/image.py]: Can you make the other returns explicit False ? We may run into weird stuff if we return None.

File: src/sagemaker/local/image.py (hunk 1 above)
Comment (ChoiByungWook) [reply src/sagemaker/local/image.py]: We set the log level to warning, so it ends up not showing up in jupyter :(.

https://github.com/aws/sagemaker-python-sdk/blob/cf652c285ca12fc0cc3c78c6126a1551b249df92/src/sagemaker/local/image.py#L45

File: src/sagemaker/local/image.py (hunk 2 above)
Comment (ChoiByungWook) [reply src/sagemaker/local/image.py]: good point.

File: src/sagemaker/local/image.py (hunk 1 above)
Comment (jesterhazy) [reply src/sagemaker/local/image.py]: that's what log level are for. why force user to see this info-level message if they don't want to?

File: src/sagemaker/local/image.py (hunk 2 above)
Comment (jesterhazy) [reply src/sagemaker/local/image.py]: remove this return. it is only used for test assert and has no other use. subprocess.check_output will throw if there is a real error, so the return value isn't even needed for tests.

File: src/sagemaker/local/image.py [hunk 3]
```diff
@@ -657,11 +660,11 @@ def _write_json_file(filename, content):
 def _ecr_login_if_needed(boto_session, image):
     # Only ECR images need login
     if not ('dkr.ecr' in image and 'amazonaws.com' in image):
-        return
+        return False
```
Comment (jesterhazy) [inline src/sagemaker/local/image.py]: remove

File: src/sagemaker/local/image.py [hunk 4]
```diff
@@ -657,11 +660,11 @@ def _write_json_file(filename, content):
...
 
     # do we have the image?
     if _check_output('docker images -q %s' % image).strip():
-        return
+        return False
```
Comment (jesterhazy) [inline src/sagemaker/local/image.py]: remove

File: tests/unit/test_image.py [hunk 5]
```diff
@@ -537,20 +537,22 @@ def test_prepare_serving_volumes_with_local_model(get_data_source_instance, sage
 
 def test_ecr_login_non_ecr():
     session_mock = Mock()
-    sagemaker.local.image._ecr_login_if_needed(session_mock, 'ubuntu')
+    result = sagemaker.local.image._ecr_login_if_needed(session_mock, 'ubuntu')
```
Comment (jesterhazy) [inline tests/unit/test_image.py]: remove return value. test the exception(s) instead

File: tests/unit/test_image.py [hunk 6]
```diff
@@ -537,20 +537,22 @@ def test_prepare_serving_volumes_with_local_model(get_data_source_instance, sage
...
+    assert result is False
 
 
 @patch('sagemaker.local.image._check_output', return_value='123451324')
 def test_ecr_login_image_exists(_check_output):
     session_mock = Mock()
 
     image = '520713654638.dkr.ecr.us-east-1.amazonaws.com/image-i-have:1.0'
-    sagemaker.local.image._ecr_login_if_needed(session_mock, image)
+    result = sagemaker.local.image._ecr_login_if_needed(session_mock, image)
```
Comment (jesterhazy) [inline tests/unit/test_image.py]: remove

File: tests/unit/test_image.py [hunk 7]
```diff
@@ -537,20 +537,22 @@ def test_prepare_serving_volumes_with_local_model(get_data_source_instance, sage
 
 def test_ecr_login_non_ecr():
     session_mock = Mock()
-    sagemaker.local.image._ecr_login_if_needed(session_mock, 'ubuntu')
+    result = sagemaker.local.image._ecr_login_if_needed(session_mock, 'ubuntu')
 
     session_mock.assert_not_called()
+    assert result is False
```
Comment (jesterhazy) [inline tests/unit/test_image.py]: remove

File: tests/unit/test_image.py [hunk 8]
```diff
@@ -577,13 +579,26 @@ def test_ecr_login_needed(check_output):
     }
     session_mock.client('ecr').get_authorization_token.return_value = response
     image = '520713654638.dkr.ecr.us-east-1.amazonaws.com/image-i-need:1.1'
-    sagemaker.local.image._ecr_login_if_needed(session_mock, image)
+    result = sagemaker.local.image._ecr_login_if_needed(session_mock, image)
```
Comment (jesterhazy) [inline tests/unit/test_image.py]: remove

File: tests/unit/test_image.py [hunk 9]
```diff
@@ -577,13 +579,26 @@ def test_ecr_login_needed(check_output):
...
 
     expected_command = 'docker login -u AWS -p %s https://520713654638.dkr.ecr.us-east-1.amazonaws.com' % token
 
     check_output.assert_called_with(expected_command, shell=True)
     session_mock.client('ecr').get_authorization_token.assert_called_with(registryIds=['520713654638'])
 
+    assert result is True
```
Comment (jesterhazy) [inline tests/unit/test_image.py]: remove. test mock was called

File: src/sagemaker/local/image.py (hunk 2 above)
Comment (ChoiByungWook) [reply src/sagemaker/local/image.py]: Don't we need this return to know whether or not to pull as defined here: https://github.com/aws/sagemaker-python-sdk/pull/455/files#diff-04b98903463aa4dcc1df3eafda085121R121

**Pull request merged**